saltgui/static/scripts/routes/home.js

```diff
@@ -75,6 +90,51 @@ class HomeRoute extends Route {
     if(this.keysLoaded && this.jobsLoaded) this.resolvePromise();
   }
 
+  _addMenu(element) {
+    var menuDropdown = this._createDiv("run-command-button", "");
+    var menuButton = this._createDiv("menu-dropdown", "&#9658;");
+    menuDropdown.appendChild(menuButton);
+    var menuDropdownContent = this._createDiv("menu-dropdown-content", "");
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x1"));
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x2"));
+    //menuDropdownContent.appendChild(this._createDiv("run-command-button", "x3"));
+    menuDropdown.appendChild(menuDropdownContent);
+    element.appendChild(menuDropdown);
+    return menuDropdownContent;
+  }
+
+  _addMenuItemAccept(hostname) {
+    var acceptButton = this._createDiv("run-command-button", "Accept&nbsp;key");
+    acceptButton.addEventListener('click', evt => {
+      this._runAcceptKey(hostname, evt);
+    });
+    return acceptButton;
+  }
+
+  _addMenuItemDelete(hostname) {
+    var deleteButton = this._createDiv("run-command-button", "Delete&nbsp;key");
+    deleteButton.addEventListener('click', evt => {
+      this._runDeleteKey(hostname, evt);
+    });
+    return deleteButton;
+  }
+
+  _addMenuItemReject(hostname) {
+    var rejectButton = this._createDiv("run-command-button", "Reject&nbsp;key");
+    rejectButton.addEventListener('click', evt => {
+      this._runRejectKey(hostname, evt);
+    });
+    return rejectButton;
+  }
+
+  _addMenuItemSyncState(hostname) {
+    var highStateButton = this._createDiv("run-command-button", "Sync&nbsp;state");
+    highStateButton.addEventListener('click', evt => {
+      this._runHighState(hostname, evt);
+    });
+    return highStateButton;
+  }
+
```

Comment:
I see a lot of duplication here. If you would create a separate class of the menu you could create a private method on this class as a helper to have less duplication.